src/execution_algorithms/ltfb/random_pairwise_exchange.cpp

```diff
@@ -307,15 +312,15 @@ void RandomPairwiseExchange::select_next(model& m,
     // FIXME (trb 03/18/21): This is ... not great. We need to
     // unravel the "fake" polymorphism in the model non-hierarchy
     // soon.
-    using DAGModel = directed_acyclic_graph_model;
-    auto& local_model = dynamic_cast<DAGModel&>(m);
-    auto& partner_dag_model = dynamic_cast<DAGModel&>(*partner_model);
-    local_model = std::move(partner_dag_model);
+    
```

Comment:
We are currently always keeping the local model:
```suggestion
    using DAGModel = directed_acyclic_graph_model;
    auto& local_model = dynamic_cast<DAGModel&>(m);
    auto& partner_dag_model = dynamic_cast<DAGModel&>(*partner_model);
    local_model = std::move(partner_dag_model);
    
```

Reply:
This isn't right. Restoring this means that `local_model`, which is an alias for `m`, will now be set to the `partner_model` (i.e., the winning model in this case). But the documentation says that it's the *losing* model that gets mutated.

In summary: the way that it's written currently, without committing @timmoon10's suggestion, the losing model, which is the local model, gets mutated and returns. This breaks the current LTFB implementation in a fairly obvious way. Merging @timmoon10's suggestion restores the normal LTFB behavior when using the `NullMutation` strategy, but breaks the documented behavior when using `ReplaceActivation`. There's clearly a deeper logic error that needs to be resolved here.

Reply:
Yes, fixed this issue. The logic now is that if a model loses in a tournament, the winning model is copied over to it and then mutated.